### ISSUE TITLE
Feature request: address-only match

### ISSUE THREAD

**Issue by skintigh** [joxeankoret/diaphora#72]
It would be great to have a fast match of just address to address (and nothing else) for transferring work form one .idb to another .idb.

This would be helpful for times when you start Ida with the wrong architecture and later need to change it, but doing so causes Ida to undefine much of your work. It could also allow two people working on the same binary to combine work.

**Comment by nirizr** [comment]
Won't 'export to IDC' work for both of these cases?

**Comment by skintigh** [comment]
I tried that and it renamed about 2/3 functions -- it lost about 600 of 1900 named functions, and about 200 unnamed functions. However, it turned almost every function into Arm when they were Thumb. Worst for me is it doesn't copy any comments, nor structures and probably other details.

Diaphora is far superior, but it's too smart for it's own good and mismatches a lot of functions. I end up with about 1000 errors that it can't rename a byte, however I'm not sure how many of those are actual problems yet, I'm still going through them.

**Comment by joxeankoret** [comment]
Uhm... you're not the only one who asked for this feature. It should be rather easy to do. I will work on it.

**Comment by joxeankoret** [comment]
NOTE: Going to be implemented for this version.

**Comment by AGG2017** [comment]
Simple address match with Diaphora is not that easy for close versions of embedded firmware with many almost equal functions. I did some tests and cannot succeed with more that 80% match for versions that are more than 95% equal. So I modified diaphora output to send the function, instruction and call graph info to MySQL server where I can process all that further. The simplest way I found for address match is initially to find 100% match for all unique clean assembly functions (with optimized cleaning procedure for the selected MCU to produce better results) which will create the basic net of the final match and then by using the call graph between them or positional search in close address window (usually +/-10%) to match the rest of the functions by using the rest of the function parameters. The result is 50-80% initial match for close versions and almost 100% match for all equal and modified functions after the final procedure where in addition to the best match also are found the modified functions, unmatched in primary and unmatched in secondary databases. Additional information that I export for each function (list of data names in the order they are used and list of their addresses) will help to create also a match of the data used. Positional search appears to be more accurate to find the changed function especially when they have changed a lot. I simplified most of the information in the database just to be fast and easy to import the instruction info from the old version to the new one for all best matched functions found (obviously there is no reason to import instruction information when the function has been modified) . The import is very simple because everything in instructions table is based on the offset from the beginning of the function they belong, so when we know the match of the addresses of the functions using the stored offset for each instruction can give you accurate match at instruction level. This instruction offset is used also to import the custom labels, comments and custom data names. It needs much more work to make it fully automated but even with some small manual work I'm completely satisfied from the result. Now there is no more wrongly matched very close functions and I can trust all the imported information. For diffing not so close versions or where is a significant change in the linking setup breaking the effectiveness of the simple positional search diaphora can produce much better result. I didn't complete my call graph search and function match. It should be in help in such cases.
So yes, simple function address match can be done but how accurate it will be using the current diaphora diffing functions is very hard to say and it depends a lot of the function contents. For very unique functions you will have good results but for many similar functions the result will be partial or with some wrong matches.
